docs/config-file/v2.rst

```diff
@@ -0,0 +1,435 @@
+Configuration File V2
+=====================
+
+Read the Docs supports configuring your documentation builds with a YAML file.
+:doc:`The Read the Docs file <index>` must be in the root directory of your project.
+
+Here is an example of how this file looks like:
+
+.. code:: yaml
+
+   # .readthedocs.yml
+
+   version: 2
+
+   python:
+     version: 3.6
+     install:
+        - method: pip
+          path: .
+
+Supported settings
+------------------
+
+.. note::
+   
+   The presence of any other key that isn't documented here will make the build to fail.
+   This is to avoid common errors.
+
+version
+~~~~~~~
+
+:Required: ``true``
+
+Example:
+
+.. code-block:: yaml
+   
+   version: 2
+
+.. warning::
+   
+   If you don't provide the version, :doc:`v1 <v1>` would be used.
+
+formats
+~~~~~~~
+
+Formats of the documentation to be built.
+
+:Type: ``list``
+:Options: ``htmlzip``, ``pdf``, ``epub``
+:Default: ``[]``
+
+Example:
+
+.. code-block:: yaml
+
+   # Default
+   formats: []
+
+.. code-block:: yaml
+
+   # Build PDF & ePub
+   formats:
+     - epub
+     - pdf
+
+.. note::
+
+   You can use the ``all`` keyword to indicate all formats.
+
+   .. code-block:: yaml
+
+      # Build all formats
+      formats: all
+
+python
+~~~~~~
+
+Configuration of the Python environment to be used.
+Example:
+
+.. code-block:: yaml
+
+   python:
+      version: 3.6
+      install:
+         - requirements: docs/requirements.txt
+         - method: pip
+           path: .
+           extra_requirements: docs
+         - method: setuptools
+           path: another/package
+      system_packages: true
+
+python.version
+``````````````
+
+The Python version (this depends on :ref:`config-file/v2:build.image`).
+
+:Type: ``number``
+:Default: ``3``
+
+python.install
```

Comment:
Not really sure if this whole section is clear. Now that we accept a list with different keys.